scripts/datascience/themes_quality/wiki_puzzle_gemini_pipeline/mp_puzzle_generation/EmbeddingCacheManager.py

```
@@ -0,0 +1,499 @@
+            mode: File open mode ('r' for read, 'w' for write)
+        """
+        lock_acquired = False
+        lock_fd = None
+        file_fd = None
+        
+        try:
+            # Create lock file if it doesn't exist
+            self.lock_file.touch(exist_ok=True)
+            
+            # Open lock file
+            lock_fd = open(self.lock_file, 'w')
```

Ruff (0.12.2)

Use a context manager for opening files

(SIM115)